### PR TITLE
Kafka: Add support for custom `command` for brokers

### PR DESCRIPTION
**Pull request by chris-vest**
**What this PR does / why we need it**:

Adds flexibility to the `command` section of the `statefulset.yaml`. My use case is that I want to prioritise the DNS name I am setting with ExternalDNS annotation on the headless service over the pod IP. As such a custom `command` is required for the brokers.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

None.

**Special notes for your reviewer**:

@benjigoldberg As discussed!

### CODE REVIEW

File: incubator/kafka/README.md [hunk 1]
```diff
@@ -69,7 +69,7 @@ following configurable parameters:
...
 | `headless.annotations`                                  | List of annotations for the headless service. https://kubernetes.io/docs/concepts/services-networking/service/#headless-services                                                            | `[]`                                                               |
 | `headless.targetPort`                                  | Target port to be used for the headless service. This is not a required value.                                                            | `nil`                                                               |
-| `headless.port`                                  | Port to be used for the headless service. https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/                                                           | `9092`                                                               |
+| `headless.port`                                  | Port to be used for the headless service.                                                           | `9092`                                                               |
```
Comment (chris-vest) [inline incubator/kafka/README.md:72]: Removing wrong link - cleaning up from my last PR when this was added!

**Comment by benjigoldberg** [comment]
@chris-vest my only concern w/ overriding the command like this is that we use interpolation on some of these variables. The way helm works I dont think they would render if a user tried to specify them in the cmd like this. Have you tried it out w/ the default cmd that exists today?

**Comment by chris-vest** [comment]
@benjigoldberg That's true! For my use case I won't be using any interpolation on the custom command, so tbh I hadn't thought about that...!

**Comment by benjigoldberg** [comment]
@chris-vest would the [overrides available for external/internal listeners](https://github.com/helm/charts/blob/master/incubator/kafka/values.yaml#L154) help? It gets set to [`advertisedListenersOverride` here](https://github.com/helm/charts/blob/master/incubator/kafka/templates/statefulset.yaml#L1) and then used as an override on [this line](https://github.com/helm/charts/blob/master/incubator/kafka/templates/statefulset.yaml#L186)

**Pull request closed without merging**